server/consumer.go

```diff
@@ -3374,6 +3374,25 @@ func (o *consumer) isFilteredMatch(subj string) bool {
 	return false
 }
 
+// Check if the candidate filter subject is equal to or a subset match
+// of one of the filter subjects.
+// Lock should be held.
+func (o *consumer) isEqualOrSubsetMatch(subj string) bool {
+	for _, filter := range o.subjf {
```

Comment:
@Jarema do we keep track if the filter has a wildcard?

Reply:
yes, the `subjectFilter` struct has bool field `hasWildcard`.

Reply:
ok, then we can skip comparison in this loop if so? Might not have any consequences, but strict equality with wildcards looks strange to me so would rather avoid here if possible.

Reply:
I've pushed a commit that implements this. It does not seem to have an effect on the tests though. There are two tests in the set where the tested and the matched subjects are equal and have a wildcard.  For both cases the result is still `true`, albeit it's probably more computationally complex to come to that decision.

If we have a consumer that filters for `foo.>` and purge the stream with the filter `foo.>`, then the consumer's sequence needs to be updated, right? Same if the consumer filters for `foo.*.bar` and we purge with `foo.*.bar`. It seems to be correct to check for equality here, even if the filter has a wildcard. But maybe I am missing cases where this doesn't hold true?